File: RewardSampleForObjC/RewardSampleForObjC/BorderButton.m
//
//  BorderButton.m
//  RewardSampleForObjC
//
//  Created by AdGeneration on 2017/02/21.
//  Copyright © 2017年 Supership Inc. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "BorderButton.h"


@interface BorderButton()

@property (nonatomic, copy) IBInspectable UIColor *textColor;
@property (nonatomic, assign) IBInspectable CGFloat cornerRadius;
@property (nonatomic, assign) IBInspectable CGFloat borderWidth;
@property (nonatomic, copy) IBInspectable UIColor  *borderColor;
@property (nonatomic, copy) IBInspectable UIColor *highlightedBackgroundColor;
@property (nonatomic, copy) IBInspectable UIColor *nonHighlightedBackgroundColor;

@end

@implementation BorderButton

- (instancetype) initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    
    if (self) {
        self.layer.cornerRadius = 5.f;
        self.layer.borderWidth = 1.f;
        self.layer.borderColor = [UIColor lightGrayColor].CGColor;
    }
    
    return self;
}

- (void) setCornerRadius:(CGFloat)cornerRadius {
    self.layer.cornerRadius = cornerRadius;
}

- (void) setBorderWidth:(CGFloat)borderWidth {
    self.layer.borderWidth = borderWidth;
}

- (void) setBorderColor:(UIColor *)borderColor {
    self.layer.borderColor = (__bridge CGColorRef _Nullable)(borderColor);
}

- (BOOL) isHighlighted {
    
    if ([super isHighlighted]) {
        self.backgroundColor = _highlightedBackgroundColor;
        self.layer.borderColor = [UIColor colorWithRed:21.0/255 green:126.0/255 blue:251.0/255 alpha:1.0].CGColor;
        
     
    } else {
        self.backgroundColor = _nonHighlightedBackgroundColor;
        self.layer.borderColor = [UIColor lightGrayColor].CGColor;
        
    }
    
    return [super isHighlighted];
}

@end

File: RewardSampleForSwift/RewardSampleForSwift/BorderButton.swift
//
//  BorderButton.swift
//  RewardSampleForSwift
//
//  Created by AdGeneration on 2017/02/21.
//  Copyright © 2017年 Supership Inc. All rights reserved.
//
// Button_Custom と同様の内容ですが、そのまま使用しているとStoryBoardが暴走したので名称を変更

import Foundation
class BorderButton: UIButton {
    
    @IBInspectable var textColor: UIColor?
    
    @IBInspectable var cornerRadius: CGFloat = 0 {
        didSet {
            layer.cornerRadius = cornerRadius
        }
    }
    
    @IBInspectable var borderWidth: CGFloat = 0 {
        didSet {
            layer.borderWidth = borderWidth
        }
    }
    
    @IBInspectable var borderColor: UIColor = UIColor.clear {
        didSet {
            layer.borderColor = borderColor.cgColor
        }
    }
    @IBInspectable var highlightedBackgroundColor :UIColor?
    @IBInspectable var nonHighlightedBackgroundColor :UIColor?
    override var isHighlighted :Bool {
        didSet {
            if isHighlighted {
                self.backgroundColor = highlightedBackgroundColor
                self.layer.borderColor = UIColor(red: 21/255, green: 126/255, blue: 251/255, alpha: 1.0).cgColor
            }
            else {
                self.backgroundColor = nonHighlightedBackgroundColor
                self.layer.borderColor = UIColor.lightGray.cgColor
            }
        }
    }
}
